v2/internal/pkg/operator/common.go

```diff
@@ -258,3 +264,213 @@ func (o OperatorCollector) prepareM2DCopyBatch(images map[string][]v2alpha1.Rela
 	}
 	return result, nil
 }
+
+func (o OperatorCollector) dispatchImagesForM2M(images map[string][]v2alpha1.RelatedImage) ([]v2alpha1.CopyImageSchema, error) {
```

Comment:
@sherine-k -  lets maybe consider refactroing this code - I see a lot of overlap for example in prepareD2MCopyBatch - but not for now , maybe in 4.19 or later

Reply:
Exactly! My goal was to introduce a new implementation, which I think is simpler.
And with the feedback we have in this PR, let's get it to a good place, so that we can remove `prepareD2MCopyBatch` and `prepareM2DCopyBatch` and replace with similar logic